torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -18,6 +18,13 @@ py::object to_py_obj(const Message& message) {
   }
 }
 
+FutureMessage::Callback wrap_callback(
+    FutureMessage& fut, const std::function<void(FutureMessage&)> py_cb) {
+  return [py_cb, &fut]() {
```

Comment:
Why do we have this wrapper? We can just convert to py::object using `to_py_obj` and pass py::object to py_cb.

Reply:
Was initially thinking about avoiding conversion when unnecessary. I can make the change if that feels more nature, and we come back to address the overhead only if we do see that in the future.